sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -600,13 +601,12 @@ class GpuSpecifiedWindowFrameMeta(
           case Literal(value, ShortType) => value.asInstanceOf[Short].toLong
           case Literal(value, IntegerType) => value.asInstanceOf[Int].toLong
           case Literal(value, LongType) => value.asInstanceOf[Long]
-          case Literal(value, CalendarIntervalType) =>
-            val interval = value.asInstanceOf[CalendarInterval]
-            if (interval.microseconds != 0 || interval.months != 0) { // DAYS == 0 is permitted.
-              willNotWorkOnGpu(s"Bounds for Range-based window frames must be specified" +
-                s" only in DAYS. Found $interval")
+          case Literal(ci: CalendarInterval, CalendarIntervalType) =>
+            if (ci.months != 0) {
+              willNotWorkOnGpu("interval months isn't supported")
             }
-            interval.days.toLong
+            // return the total microseconds
+            ci.days * TimeUnit.DAYS.toMicros(1) + ci.microseconds
```

Comment:
This calculation can overflow. In the normal SQL case it should be very unlikely to happen, but we should still detect it before it happens and fall back to the CPU.

Reply:
Yeah, the overflow checking is quite necessary, Thx for the commenting. Done